### PR TITLE
Improve vacancy page layout

### LINKED ISSUE #1937: Improve vacancy page lay-out
### Is your feature request related to a problem? Please describe.
We need a better lay-out for the vacancies.

### Describe the solution you'd like
I think it would be better if we displayed the vacancies in a simple list of items like this: 
```
[...partner image...] <title> 
<partner name> - <partner location> <optional tags> - <vacancy category>
<summary>
<learn more>
```

So for example:
```
[nedap.png] (junior) software engineer
Nedap - Groenlo - Young Professional
Nedap Light Controls realiseert duurzame verlichtingsoplossingen via zelf ontwikkelde hard- en software. Met ons product Luxon kunnen onze klanten hu…
Learn more


[infocaster.png] junior .NET (C#) ontwikkelaar
InfoCaster - Arnhem - .NET - Young Professional
Geef je carrière als developer een kickstart en werk samen met het .NET-team van Infocaster aan complexe vraagstukken voor klanten. Je realise…
Learn more
```

### Motivation
The vacancies page will be easier to read and more students will use the vacancies.

### Additional context
See #1845

### PR DESCRIPTION
**Pull request by wdoeland**
Closes #1937 

### Summary
Improved the vacancy layout

### How to test
Steps to test the changes you made:
1. Go to `/career/vacancies`
2. Check if you can see locations and tags
3. Maybe add a location and tags to the vacancies in the admin panel
4. Check again

Screen Shot 2021-09-20 at 21 41 01

### CODE REVIEW

File: website/partners/models.py [hunk 1]
```diff
@@ -138,6 +138,8 @@ class Vacancy(models.Model):
     link = models.CharField(
         _("link"), max_length=255, blank=True, validators=[URLValidator()]
     )
+    location = models.CharField(_("location"), max_length=255, default="")
+    keywords = models.TextField(_("keywords"), default="")
```
Comment (se-bastiaan) [inline website/partners/models.py:142]: This field requires a help text.

File: website/partners/models.py [hunk 2]
```diff
@@ -138,6 +138,8 @@ class Vacancy(models.Model):
     link = models.CharField(
         _("link"), max_length=255, blank=True, validators=[URLValidator()]
     )
+    location = models.CharField(_("location"), max_length=255, default="")
```
Comment (se-bastiaan) [inline website/partners/models.py:141]:
```suggestion
    location = models.CharField(_("location"), max_length=255, null=True, blank=True)
```

**Review by se-bastiaan — CHANGES_REQUESTED** [review]
Something is going wrong with the tests and codestyle, idea looks good!

**Comment by wdoeland** [comment]
I can't really fix the migrations. I deleted them again and I'm still having the same issue. It says there's conflicting migrations but one of the migrations (0020_auto_20211002_0002) is not even in this branch.

**Comment by se-bastiaan** [comment]
Did you rebase? Sounds like you didn't and that is what it is expecting from you.

**Review by se-bastiaan — APPROVED** [review]
Rebased it for you

**Pull request merged**